sdk/playwrighttesting/microsoft-playwright-testing/test/common/playwrightServiceConfig.spec.ts

```diff
@@ -98,11 +98,42 @@ describe("PlaywrightServiceConfig", () => {
     expect(playwrightServiceConfig.exposeNetwork).to.equal(
       DefaultConnectOptionsConstants.DEFAULT_EXPOSE_NETWORK,
     );
-    expect(process.env[ServiceEnvironmentVariable.PLAYWRIGHT_SERVICE_RUN_ID]).to.equal(
+    expect(process.env[InternalEnvironmentVariables.MPT_SERVICE_RUN_ID]).to.equal(
       playwrightServiceConfig.runId,
     );
 
-    delete process.env[ServiceEnvironmentVariable.PLAYWRIGHT_SERVICE_RUN_ID];
+    delete process.env[InternalEnvironmentVariables.MPT_SERVICE_RUN_ID];
     delete process.env[ServiceEnvironmentVariable.PLAYWRIGHT_SERVICE_OS];
   });
+  it("should set runId from options if provided and environment variable is not set", () => {
+    const playwrightServiceConfig = new PlaywrightServiceConfig();
+    playwrightServiceConfig.setOptions({
+      runId: "custom-run-id",
+    });
+    expect(playwrightServiceConfig.runId).to.equal("custom-run-id");
+    expect(process.env[InternalEnvironmentVariables.MPT_SERVICE_RUN_ID]).to.equal("custom-run-id");
+    delete process.env[InternalEnvironmentVariables.MPT_SERVICE_RUN_ID];
+  });
+  it("should generate runId if not provided in options and environment variable is not set", () => {
+    const runId = getAndSetRunId();
+    const playwrightServiceConfig = new PlaywrightServiceConfig();
+    playwrightServiceConfig.setOptions({
+      runId: runId,
+    });
+    expect(playwrightServiceConfig.runId).to.be.a("string");
+    expect(playwrightServiceConfig.runId).to.equal(runId);
+    expect(process.env[InternalEnvironmentVariables.MPT_SERVICE_RUN_ID]).to.equal(runId);
+    delete process.env[InternalEnvironmentVariables.MPT_SERVICE_RUN_ID];
+  });
+  it("should use runId from environment variable if already set", () => {
+    process.env[InternalEnvironmentVariables.MPT_SERVICE_RUN_ID] = "existing-run-id";
+    const playwrightServiceConfig = new PlaywrightServiceConfig();
+    playwrightServiceConfig.setOptions();
```

Comment:
NIT - You should pass runid here as well and ensure this is not being used to set the environment variable